src/solvers/dg_simplices/dg.jl

```diff
@@ -1,14 +1,17 @@
 # !!! warning "Experimental features"
 
-# Todo: simplices. Investigate replacing `mul!` by `Octavian.matmul!` for additional speedup.
-
 # out <- A*x
 mul_by!(A) = let A = A 
   @inline (out, x)->matmul!(out, A, x) 
 end
 
-# Don't use `matmul!` for the following 2 functions: 5-arg `matmul!` hangs when applied to vector 
-# arguments. See https://github.com/JuliaLinearAlgebra/Octavian.jl/issues/103. 
+# specialize for SBP operators since matmul! doesn't work for `UniformScaling` types
+mul_by!(A::UniformScaling) = let A = A
+  @inline (out, x)->out .= x
+end
```

Comment:
That's problematic:
```juliajulia> using LinearAlgebra

julia> typeof(I)
UniformScaling{Bool}

julia> UniformScaling(2.0)
UniformScaling{Float64}
2.0*I
```
It handles the situation where you pass `I` correctly but will be false in general... Maybe restrict it to `UniformScaling{Bool}` and hope that nobody uses
```julia
julia> UniformScaling(false)
UniformScaling{Bool}
false*I
```
Alternatively, you could just use `mul!` here, which might be slightly more costly due to an additional multiplication but would be correct in general.

Reply:
Hm, good point. I prefer the safer `mul!` fallback for now. 

This can also be avoided by dispatching based on `DG` types. Evaluating the solution at quadrature points requires a matrix multiply, as does performing an L2 projection. For collocated discretizations like DGSEM or nodal SBP, I just set the relevant matrices to `I` as a way to "skip" matrix multiplication. 

How about we use the `mul!` fallback for now, and I add this to the todo-list https://github.com/trixi-framework/Trixi.jl/issues/675?

Reply:
Sounds good :+1: